test/pmi_client.c

```diff
@@ -334,30 +328,6 @@ static int test_item5(void)
 static int test_item6(void)
 {
     int rc = 0;
-    char val[100];
-    const char *tkey = __func__;
-    const char *tval = __FILE__;
-
-    if (PMI_SUCCESS != (rc = PMI_KVS_Put(jobid, tkey, tval))) {
-        log_fatal("PMI_KVS_Put %d\n", rc);
-        return rc;
-    }
-
-    if (PMI_SUCCESS != (rc = PMI_KVS_Get(jobid, tkey, val, sizeof(val)))) {
```

Comment:
I guess that it is not clear that my comment below is targeted here (I replied to the email instead of writing through the web).

Here is what I'm trying to say. As far as I recall doing Get without Commit/Barrier sequence is leading to the unpredictable results. 

This is what I'm observing now.
- In case of message-based server-client exchange we 
a) store the data locally in the hash table:
https://github.com/pmix/master/blob/master/src/client/pmix_client.c#L566
b) store it in the caches pending Commit call:
https://github.com/pmix/master/blob/master/src/client/pmix_client.c#L571:L595

- In case of dstore we only store it in the cache, so this test is failing. But I guess that according to the PMI1 specification this failure is expected. 

So that's why I'm deleting this test.